### PR TITLE
[sdl2] Update to 2.0.16

### PR DESCRIPTION
**Pull request by RT2Code**
Update the SDL2 port from 2.0.14 to 2.0.16 :
https://github.com/libsdl-org/SDL/releases/tag/release-2.0.16

I removed three files :
- 0003-sdl2-fix-uwp-build.patch because it was merged upstream : https://github.com/libsdl-org/SDL/commit/d54b125a860e83dac32efc0029e0055f0d1435c3
- 0004-sdl2-macos-thread-detection-fix.patch because they fixed pthread detection on MacOS : https://github.com/libsdl-org/SDL/commit/89a2980a6d55ace688ffca1ecef45c0eb432af2a
- vcpkg-cmake-wrapper.cmake because it was not compatible with the changes they made to the SDL2Config.cmake file. I replaced it with a new patch borrowed from this PR : https://github.com/libsdl-org/SDL/pull/4502

I was able to build with the Vulkan feature with success on Windows.

### CODE REVIEW

File: ports/sdl2/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "sdl2",
-  "version-string": "2.0.14",
-  "port-version": 4,
+  "version-string": "2.0.16",
```
Comment (JonLiu1993) [inline ports/sdl2/vcpkg.json:3]:
```suggestion
  "version-semver": "2.0.16",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/sdl2/vcpkg.json (hunk 1 above)
Comment (mathisloge) [reply ports/sdl2/vcpkg.json:3]: is sdl2 following the semver versioning? Couldn't find any mention of that. (But it looks like it is following it)

Comment (Yamakaky) [reply ports/sdl2/vcpkg.json:3]: They use the patch version when adding backward-compatible features so I'd say it's not really following semver.

Comment (RT222) [reply ports/sdl2/vcpkg.json]: Yes, [according to this repo](https://github.com/Lokathor/fermium), SDL2 doesn't follow semver :

> SDL2 does not follow SemVer.
> 
> Under SDL2's version scheme, the numbers are what SemVer would call major.major.minor. In other words, 2.0.10 is a compatible API upgrade from 2.0.9, and 2.1.0 is a breaking upgrade from 2.0.9.

**Comment by BillyONeal** [comment]
Pushed merge with master because the VMs for the running validation build are being removed.

**Comment by RT222** [comment]
Finally, the commit history is a mess, but I think I got it right. On UWP, I had to remove the static libs definition in the pkgconfig file and to disable the SDL2main target lookup in SDL2Config.cmake because we don't ship the sdl2main on this platform.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
FeATURE vulkan tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Pull request merged**